authorize/evaluator/evaluator.go

```
@@ -4,48 +4,320 @@ package evaluator
+	Session                  RequestSession `json:"session"`
+	IsValidClientCertificate bool           `json:"is_valid_client_certificate"`
+}
+
+func (e *Evaluator) newInput(req *Request, isValidClientCertificate bool) *input {
+	i := new(input)
+	i.DataBrokerData = req.DataBrokerData
+	i.HTTP = req.HTTP
+	i.Session = req.Session
+	i.IsValidClientCertificate = isValidClientCertificate
+	return i
+}
+
+type (
+	// Request is the request data used for the evaluator.
+	Request struct {
+		DataBrokerData DataBrokerData `json:"databroker_data"`
+		HTTP           RequestHTTP    `json:"http"`
+		Session        RequestSession `json:"session"`
+	}
+
+	// RequestHTTP is the HTTP field in the request.
+	RequestHTTP struct {
+		Method            string            `json:"method"`
+		URL               string            `json:"url"`
+		Headers           map[string]string `json:"headers"`
+		ClientCertificate string            `json:"client_certificate"`
+	}
+
+	// RequestSession is the session field in the request.
+	RequestSession struct {
+		ID                string   `json:"id"`
+		ImpersonateEmail  string   `json:"impersonate_email"`
+		ImpersonateGroups []string `json:"impersonate_groups"`
+	}
+)
+
+// Result is the result of evaluation.
+type Result struct {
+	Status    int
+	Message   string
+	SignedJWT string
+}
+
+func getAllowVar(vars rego.Vars) bool {
+	result, ok := vars["result"].(map[string]interface{})
+	if !ok {
+		return false
+	}
+
+	allow, ok := result["allow"].(bool)
+	if !ok {
+		return false
+	}
+	return allow
+}
+
+func getDenyVar(vars rego.Vars) []Result {
+	result, ok := vars["result"].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	denials, ok := result["deny"].([]interface{})
+	if !ok {
+		return nil
+	}
+
+	var results []Result
+	for _, denial := range denials {
+		denial, ok := denial.([]interface{})
+		if !ok || len(denial) < 2 {
+			continue
+		}
+
+		status, err := strconv.Atoi(fmt.Sprint(denial[0]))
+		if err != nil {
```

Could we return this error, or warn about it. When rego changes the result structure, this can lead to hidden bug or difficult to debug code.